pkg/url/url.go

```diff
@@ -100,6 +103,79 @@ func GetIngress(kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInf
 	return iextensionsv1.Ingress{}, errors.New(fmt.Sprintf("the url %v does not exist", urlName))
 }
 
+// GetContainer returns Docker URL definition for given URL name
+func GetContainerURL(envSpecificInfo *envinfo.EnvSpecificInfo, urlName string, componentName string) (URL, error) {
+	client, err := lclient.New()
+	if err != nil {
+		return URL{}, err
+	}
+	localURLs := envSpecificInfo.GetURL()
+	containers, err := dockerutils.GetComponentContainers(*client, componentName)
+	if err != nil {
+		return URL{}, errors.Wrap(err, "unable to get component containers")
+	}
+	for _, c := range containers {
+		containerJSON, err := client.Client.ContainerInspect(client.Context, c.ID)
```

Comment:
it seems we are calling inspect on the same container ids here https://github.com/openshift/odo/pull/3086/files#diff-1e4fc401677b064040ea6bf41227b6acR154. Instead loop and store it once in a map map[id]containerjson and fetch it from there.

Reply:
Updated